features/testbot/matching.feature

```diff
@@ -169,3 +169,47 @@ Feature: Basic Map Matching
         When I match I should get
             | trace | matchings | geometry                                   |
             | abd   | abd       | 1,1,1.000089,1,1.000089,1,1.000089,0.99991 |
+
+
+    # Regression test 1 for issue 3176
+    Scenario: Testbot - multiuple segments: properly expose OSM IDs
+        Given the query options
+            | annotations | true    |
+
+        Given the node map
+            """
+            a-1-b--c--d--e--f-2-g
+            """
+
+        And the ways
+            | nodes | oneway |
+            | ab    | no     |
+            | bc    | no     |
+            | cd    | no     |
+            | de    | no     |
+            | ef    | no     |
+            | fg    | no     |
+
+        When I match I should get
+            | trace | OSM IDs       |
+            | 12    | 1,2,3,4,5,6,7 |
```

Comment:
If we're going to depend on the node ID values in the tests, we should make sure they're stable.  I suspect the results are deterministic right now, but the behaviour is implied.

Looks like there is *some* logic to generate consistent IDs here:
https://github.com/Project-OSRM/osrm-backend/blob/master/features/support/data.js#L43

But the logic seems incomplete.

@TheMarex if you don't have time to clear this up, I can look at adding it to this PR today.  I think we should fix this at the same time.